internal/graph/check.go

```diff
@@ -323,7 +323,7 @@ func (c *LocalChecker) ResolveCheck(
 	ctx context.Context,
 	req *ResolveCheckRequest,
 ) (*ResolveCheckResponse, error) {
-	ctx, span := tracer.Start(ctx, "ResolveCheck")
+	ctx, span := tracer.Start(ctx, "checkRewrite")
 	defer span.End()
```

Comment:
I realize the span name is not consistent but the function name, but this is the right name from a customer perspective. `ResolveCheck` calls `checkRewrite`. I think we need to do some name changes after this PR. `ResolveCheck` is called both by `ListObjects` and `Check` AND within this file, which is confusing

Reply:
Yep, that's purposeful. `ResolveCheck` recurses on itself (or dispatches/delegates to itself) because the composition is meant to be able to delegate the resolution to any implementation of a `CheckResolver`, which could be a remote `CheckResolver` (dispatch sub-problems to a peer or to a cached local resolver etc..).